access/validator.go

```diff
@@ -259,6 +264,19 @@ func (v *TransactionValidator) checkAddresses(tx *flow.TransactionBody) error {
 	return nil
 }
 
+// every key (account, key index combination) can only be used once for signing
+func (v *TransactionValidator) checkSignatureDuplications(tx *flow.TransactionBody) error {
+	observedSigs := make(map[string]bool)
+	for _, sig := range append(tx.PayloadSignatures, tx.EnvelopeSignatures...) {
+		keyStr := fmt.Sprintf("%s-%d", sig.Address.String(), sig.KeyIndex)
```

Comment:
eventually we'll probably want this to be a function off the signature type, as it'd be much better to do something like `sig.Name`

Reply:
updated at 14c0c39